app/helpers/application_helper/button/basic.rb

```diff
@@ -0,0 +1,25 @@
+class ApplicationHelper::Button::Basic < Hash
+  def initialize(view_context, view_binding, instance_data, props)
+    @view_context  = view_context
+    @view_binding  = view_binding
+
+    props.each { |k, v| self[k] = v }
+
+    instance_data.each do |name, value|
+      instance_variable_set(:"@#{name}", value)
+    end
+  end
+
+  def calculate_properties
+    props["enabled"] = "false" if disabled?
```

Comment:
Well I hope there is a chance. But it's not a simple change.

I cannot simply turn everything to HashWithIndifferentAccess or turn all keys to symbols as we have the same names used as string and symbols at some places.

And unfortunatelly we also have places that change the input hash not the output one.

My plan is to convert the special cases and then move the more generic code too. The key mess should be addressed somewhere on the process.